R/tar-terra-rast.R

```diff
@@ -87,3 +95,44 @@ tar_terra_rast <- function(name,
         cue = cue
     )
 }
+
+#' @param filetype File format expressed as GDAL driver names passed to `terra::writeRaster()`
+#' @param gdal GDAL driver specific datasource creation options passed to `terra::writeRaster()`
+#' @param ... Additional arguments not yet used
+#' @noRd
+create_format_terra_raster <- function(filetype, gdal, ...) {
+
+    if (!requireNamespace("terra")) {
+        stop("package 'terra' is required", call. = FALSE)
+    }
+
+    # get list of drivers available for writing depending on what the user's GDAL supports
+    drv <- terra::gdal(drivers = TRUE)
+    drv <- drv[drv$type == "raster" & grepl("write", drv$can), ]
+
+    filetype <- match.arg(filetype, drv$name)
+
+    if (is.null(filetype)) {
+        filetype <- "GTiff"
+    }
```

Comment:
I included it here in case the format method was called without using `tar_terra_rast()`, a bit of a moot point as the `create_format...()` method is not exported.

I don't really have strong opinions on whether this needs to be be in there at all, but may be a good idea to discuss default behaviors. It would be simple enough to specify the default in the function definition and remove the conditional.

Reply:
Sure, I think that this all looks good to me, and discussing good defaults can go as a separate issue, as you note, there might be some interest in the user to set some global option or something - since this goes beyond the scope of this PR I think an issue is a good place to discuss :)